src/main/java/fi/metatavu/metaform/server/rest/V1ApiImpl.java

```diff
@@ -263,6 +269,29 @@ public Response createMetaform(@Valid Metaform payload) {
     return createOk(metaformTranslator.translateMetaform(metaform));
   }
 
+  @Override
+  public Response createMetaformVersion(UUID metaformId, @Valid MetaformVersion payload) {
+    if (!isRealmMetaformAdmin()) {
+      return createForbidden(YOU_ARE_NOT_ALLOWED_TO_CREATE_METAFORM_VERSIONS);
+    }
+
+    fi.metatavu.metaform.server.persistence.model.Metaform foundMetaform = metaformController.findMetaformById(metaformId);
+
+    if (foundMetaform == null) {
+      return createNotFound(NOT_FOUND_MESSAGE);
+    }
+
+    fi.metatavu.metaform.server.persistence.model.MetaformVersion metaformVersion =
+            metaformVersionController.create(
+                    foundMetaform,
+                    payload.getType(),
+                    payload.getData(),
+                    getLoggerUserId()
+          );
+    if (metaformVersion == null) return createBadRequest("Invalid version data");
+    return createOk(metaformVersionTranslator.translateMetaformVersion(metaformVersion));
```

Comment:
It would be better to use custom error and capture it here? (eg. MalformedVersionJsonException)